### PR TITLE
Enable Jballerina-unit-tests

### PR DESCRIPTION
**Pull request by m36dot**
## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by SupunS** [comment]
@m36dot The PR title and some of the commit messages are not according to the standards we follow. Can you please fix those?

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java [hunk 1]
```diff
@@ -3348,7 +3348,16 @@ private STNode parseIndividualRecordField(STNode metadata, STNode readOnlyQualif
      * @return Type reference node
      */
     private STNode parseTypeReference() {
-        return parseQualifiedIdentifier(ParserRuleContext.TYPE_REFERENCE, false);
+        STNode typeReference = parseTypeDescriptor(ParserRuleContext.TYPE_REFERENCE);
+        if (typeReference.kind == SyntaxKind.SIMPLE_NAME_REFERENCE ||
+            typeReference.kind == SyntaxKind.QUALIFIED_NAME_REFERENCE) {
+            return typeReference;
+        }
+        STNode dummy = STNodeFactory.createSimpleNameReferenceNode(
```
Comment (SupunS) [inline compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java:3356]: can we use a better name instead of `dummy`?

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java [hunk 2]
```diff
@@ -1944,6 +1944,9 @@ private Result seekMatchInExpressionRhs(int lookahead, int currentDepth, int cur
             nextContext = ParserRuleContext.MATCH_BODY;
         } else if (parentCtx == ParserRuleContext.MATCH_BODY) {
             nextContext = ParserRuleContext.RIGHT_DOUBLE_ARROW;
+        } else if (parentCtx == ParserRuleContext.TYPE_REFERENCE) {
+            endContext();
+            nextContext = ParserRuleContext.SEMICOLON;
```
Comment (SupunS) [inline compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java:1949]: An expression cannot exist in a `TYPE_REFERENCE` context. Root cause might be somewhere else. Shall we try to find and fix that?

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationAttachmentPointTest.java [hunk 3]
```diff
@@ -288,28 +288,9 @@ public void testInvalidAttachmentOnVar() {
...
-        validateError(compileResult, index++, "annotation 'v6' is not allowed on var", line += 3, 13);
-        validateError(compileResult, index++, "annotation 'v7' is not allowed on var", line += 3, 13);
-        validateError(compileResult, index++, "annotation 'v8' is not allowed on var", ++line, 13);
-        validateError(compileResult, index++, "annotation 'v9' is not allowed on var", line += 3, 13);
-        validateError(compileResult, index++, "annotation 'v10' is not allowed on var", line += 3, 13);
-        validateError(compileResult, index++, "annotation 'v12' is not allowed on var", line += 3, 13);
-        validateError(compileResult, index++, "annotation 'v13' is not allowed on var", line += 3, 13);
-        validateError(compileResult, index, "annotation 'v15' is not allowed on var", line + 3, 13);
-    }
-
```
Comment (SupunS) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationAttachmentPointTest.java:309]: Any reason why are these tests removed?

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/balo/record/ClosedRecordTypeReferenceTest.java [hunk 4]
```diff
@@ -54,20 +54,20 @@ public void setup() {
...
-        BAssertUtil.validateError(negative, index++, "incompatible types: 'byte' is not a record", 49, 6);
-        BAssertUtil.validateError(negative, index++, "incompatible types: 'json' is not a record", 50, 6);
-        BAssertUtil.validateError(negative, index++, "incompatible types: 'xml' is not a record", 51, 6);
+        BAssertUtil.validateError(negative, index++, "only type reference allowed here", 45, 6);
+        BAssertUtil.validateError(negative, index++, "only type reference allowed here", 46, 6);
+        BAssertUtil.validateError(negative, index++, "only type reference allowed here", 47, 6);
+        BAssertUtil.validateError(negative, index++, "only type reference allowed here", 48, 6);
+        BAssertUtil.validateError(negative, index++, "only type reference allowed here", 49, 6);
+        BAssertUtil.validateError(negative, index++, "only type reference allowed here", 50, 6);
+        BAssertUtil.validateError(negative, index++, "only type reference allowed here", 51, 6);
```
Comment (SupunS) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/balo/record/ClosedRecordTypeReferenceTest.java:70]: should it be `only type references are allowed as type inclusions`?

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java (hunk 1 above)
Comment (m36dot) [reply compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java:3356]: sure

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationAttachmentPointTest.java (hunk 3 above)
Comment (m36dot) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationAttachmentPointTest.java:309]: yes, the syntax is correct. the test seems to be wrong.

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/balo/record/ClosedRecordTypeReferenceTest.java (hunk 4 above)
Comment (m36dot) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/balo/record/ClosedRecordTypeReferenceTest.java:70]: I see, I ll change the error message.

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java (hunk 2 above)
Comment (m36dot) [reply compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java:1949]: I see, I ll try and find that.

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationAttachmentPointTest.java (hunk 3 above)
Comment (SupunS) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationAttachmentPointTest.java:309]: I think the test case is correct. We have to find and fix the root cause

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java (hunk 2 above)
Comment (m36dot) [reply compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java:1949]: since we are allowing all type descriptors inside type-reference the error handler goes into expressions as well

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationAttachmentPointTest.java (hunk 3 above)
Comment (m36dot) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationAttachmentPointTest.java:309]: yep I ll get that fixed

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationAttachmentPointTest.java [hunk 5]
```diff
@@ -82,59 +82,59 @@ public void testInvalidAttachmentOnObjectMethodDefinition() {
         int index = 25;
         int line = 117;
         validateError(compileResult, index++, "annotation 'v1' is not allowed on object_method, function",
-                      line, 5);
+                line, 5);
```
Comment (SupunS) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationAttachmentPointTest.java:85]: Shall we revert these changes if they are not needed? Should wee add that missing one too?

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationAttachmentPointTest.java [hunk 6]
```diff
@@ -29,15 +29,15 @@
...
+@Test(groups = { "disableOnOldParser" })
 public class AnnotationAttachmentPointTest {
 
     private CompileResult compileResult;
 
     @BeforeClass
     public void setup() {
         compileResult = BCompileUtil.compile("test-src/annotations/annot_attachments_negative.bal");
-        Assert.assertEquals(compileResult.getErrorCount(), 244);
+        Assert.assertEquals(compileResult.getErrorCount(), 245);
```
Comment (SupunS) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationAttachmentPointTest.java:40]: So theres an additional error being logged - but I don't see any additional assert being added. Should we add that missing one too?

File: language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/signature/SignatureHelpTest.java [hunk 7]
```diff
@@ -174,7 +174,8 @@ public void test(String config, String source)
                 {"exprChecking1.json", "functionsAndTypeDefs.bal"},
                 {"exprChecking2.json", "functionsAndTypeDefs.bal"},
                 {"exprTrap.json", "functionsAndTypeDefs.bal"},
-                {"exprServiceConstructor1.json", "functionsAndTypeDefs.bal"},
+                //disabled since giving function pos from function keyword breaks this.
+ //               {"exprServiceConstructor1.json", "functionsAndTypeDefs.bal"},
```
Comment (SupunS) [inline language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/signature/SignatureHelpTest.java:178]: Have you created an issue for these disabled tests? 
If not, shall we create a one?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java [hunk 8]
```diff
@@ -1133,12 +1138,33 @@ public BLangNode transform(FunctionDefinitionNode funcDefNode) {
                 funcDefNode.functionSignature(), funcDefNode.functionBody(), funcDefNode.transactionalKeyword());
 
         bLFunction.annAttachments = applyAll(funcDefNode.metadata().annotations());
-        bLFunction.pos = getPosition(funcDefNode);
+        bLFunction.pos = getPositionForFuncDefNode(funcDefNode);
+
+
```
Comment (SupunS) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:1143]: extraneous new lines?

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationAttachmentPointTest.java (hunk 6 above)
Comment (m36dot) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationAttachmentPointTest.java:40]: yea I ll add that additional assert, in the same position 2 errors are being thrown one by parser and one by sem Analyzer

File: language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/signature/SignatureHelpTest.java (hunk 7 above)
Comment (m36dot) [reply language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/signature/SignatureHelpTest.java:178]: yes I did create an issue for this -> https://github.com/ballerina-platform/ballerina-lang/issues/24818

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java (hunk 8 above)
Comment (m36dot) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:1143]: will remove

**Comment by m36dot** [comment]
replaced this with a new pull request, https://github.com/ballerina-platform/ballerina-lang/pull/24868
